core/shared/src/main/scala/zio/ConfigProvider.scala

```diff
@@ -330,6 +310,103 @@ object ConfigProvider {
     }
   }
 
+  trait IndexedFlat extends Flat {
+    import IndexedFlat._
+
+    def enumerateChildrenIndexed(path: ConfigPath)(implicit trace: Trace): IO[Config.Error, Set[ConfigPath]]
+
+    def loadIndexed[A](path: ConfigPath, config: Config.Primitive[A], split: Boolean)(implicit
+      trace: Trace
+    ): IO[Config.Error, Chunk[A]]
+
+    final def enumerateChildren(path: Chunk[String])(implicit trace: Trace): IO[Config.Error, Set[String]] =
+      enumerateChildrenIndexed(ConfigPath.fromPath(path)).map(_.map(_.last.value))
+
+    final def load[A](path: Chunk[String], config: Config.Primitive[A])(implicit
+      trace: Trace
+    ): IO[Config.Error, Chunk[A]] =
+      load(path, config, true)
+
+    final override def load[A](path: Chunk[String], config: Config.Primitive[A], split: Boolean)(implicit
+      trace: Trace
+    ): IO[Config.Error, Chunk[A]] =
+      loadIndexed(ConfigPath.fromPath(path), config, split)
+
+    final def loadIndexed[A](path: ConfigPath, config: Config.Primitive[A])(implicit
+      trace: Trace
+    ): IO[Config.Error, Chunk[A]] =
+      loadIndexed(path, config, true)
+  }
+
+  object IndexedFlat {
+
+    type ConfigPath = Chunk[KeyComponent]
```

Comment:
👍  So this is back without backward compatibility issues now. Am I right?

Reply:
Yes we are just implementing an interface.